src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -77,74 +77,88 @@ public async Task<JObject> GetValueFromObject(JToken objRet, CancellationToken t
             return null;
         }
 
-        public async Task<JObject> TryToRunOnLoadedClasses(string varName, CancellationToken token)
+        public async Task<(JObject containerObject, string remaining)> ResolveStaticMembersInStaticTypes(string varName, CancellationToken token)
         {
             string classNameToFind = "";
             string[] parts = varName.Split(".");
-            var typeId = -1;
-            foreach (string part in parts)
+            var store = await proxy.LoadStore(sessionId, token);
+            var methodInfo = context.CallStack.FirstOrDefault(s => s.Id == scopeId)?.Method?.Info;
+
+            int typeId = -1;
+            for (int i = 0; i < parts.Length; i++)
             {
+                string part = parts[i].Trim();
+
                 if (classNameToFind.Length > 0)
                     classNameToFind += ".";
-                classNameToFind += part.Trim();
+                classNameToFind += part;
+
                 if (typeId != -1)
                 {
-                    var fields = await context.SdbAgent.GetTypeFields(typeId, token);
-                    foreach (var field in fields)
-                    {
-                        if (field.Name == part.Trim())
-                        {
-                            var isInitialized = await context.SdbAgent.TypeIsInitialized(typeId, token);
-                            if (isInitialized == 0)
-                            {
-                                isInitialized = await context.SdbAgent.TypeInitialize(typeId, token);
-                            }
-                            var valueRet = await context.SdbAgent.GetFieldValue(typeId, field.Id, token);
-                            return await GetValueFromObject(valueRet, token);
-                        }
-                    }
-                    var methodId = await context.SdbAgent.GetPropertyMethodIdByName(typeId, part.Trim(), token);
-                    if (methodId != -1)
-                    {
-                        using var commandParamsObjWriter = new MonoBinaryWriter();
-                        commandParamsObjWriter.Write(0); //param count
-                        var retMethod = await context.SdbAgent.InvokeMethod(commandParamsObjWriter.GetParameterBuffer(), methodId, "methodRet", token);
-                        return await GetValueFromObject(retMethod, token);
-                    }
+                    string remaining = null;
+                    JObject memberObject = await FindStaticMemberInType(part, typeId);
+                    if (memberObject != null && i < parts.Length - 1)
+                        remaining = string.Join('.', parts[(i + 1)..]);
+
+                    if (memberObject != null || remaining != null)
```

Comment:
In the case of nested static type we have both variables == null. However, we still should look for the value using FindStaticTypeId, so we cannot return yet.

Reply:
Ah, I see. It looks confusing because `remaining` will be !null only when `memberObject` is !null. Maybe change it like:

```csharp
                if (typeId != -1)
                {
                    // Found a type, looking for a member named `part` in it

                    JObject memberObject = await FindStaticMemberInType(part, typeId);
                    if (memberObject != null)
                    {
                        string remaining = null;
                        if (i < parts.Length - 1)
                            remaining = string.Join('.', parts[(i + 1)..]);

                        return (memberObject, remaining);
                    }

                    // Didn't find a member named `part` in `typeId`.
                    // Could be a nested type. Let's continue the search
                    // with `part` added to the type name

                    typeId = -1;
                }

                if (classNameToFind.Length > 0)
                    classNameToFind += ".";
                classNameToFind += part;
```